[20261] [Details Pane] superfluous warning on page leave

campreb opened the pull request:
Warning message was being displayed due to a form submit event not being triggered in [this file](https://github.com/opf/openproject/blob/dev/app/assets/javascripts/warn_leaving_unsaved.js#L33-L35).
I've changed the trigger for the createComment method to be form submit, rather than the button click.

Commit:
Move activity createComment trigger to form submit.

Move how activity comments get triggered to the form submit rather than button click to allow for onbeforeunload events to be properly handled.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://campreb-bug-20261-superfluous-warning.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://campreb-bug-20261-superfluous-warning.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

campreb commented:
Just after some suggestions on how best to test this change, should I look at creating some integration tests for the activity comment form?

floriank commented:
I restarted the protractor and karma tests (we have one or two flickering tests at the moment).

New integration tests are very much appreciated, but please note that you might have to mock out the backend :wink:

floriank commented:
>  activityCommentDirective activity comments with comment link in work package does prevent double posts FAILED

This one actually fails (it's in the karma suite).

Karma suite should executable via 

```
./node_modules/.bin/karma start
```

in `frontend`

campreb commented:
@floriank yeah noticed that too, looking into it now

Commit:
Make onbeforeunload event not fire when angular form is submitted.

- Revert using the ng-submit for the activity comments form.
- Listen to form button click to remove `changed` data attribute and prevent warning.

Commit:
add Capybara specs for navigation warnings from the activity comment form

floriank commented:
I restarted the jobs. @campreb would you mind fixing the style notifications by @houndci ?

Commit:
fix ruby style in integration test

floriank commented:
Thanks!
